code-challenges.test.js

```diff
@@ -15,40 +15,93 @@
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
-
-
+describe("code", () => {
+  it("converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0", () => {
+    const secretCodeWord1 = "Lackadaisical"
+    const secretCodeWord2 = "Gobbledygook"
+    const secretCodeWord3 = "Eccentric"
+    expect(code(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+    expect(code(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+    expect(code(secretCodeWord3)).toEqual("3cc3ntr1c")
+  })
+})
 
 // b) Create the function that makes the test pass.
 
-
+//create result array
+//split input string into an array of letters and iterate over that array
+//lowercase character before checking if it is equal to lowercase input character
+//if the letter is one in the code, change it to the corresponding number
+//push each character into the result array
+//return result array joined on an emptry string
+
+const code = (str) => {
+ let result = []
+ let lettersArr = str.split('')
+ for (let i = 0; i < lettersArr.length; i++) {
+   let currentLetter = lettersArr[i]
+   let currentLowercase = lettersArr[i].toLowerCase()
+   if (currentLowercase === 'a') {
+     currentLetter = 4
+   } else  if (currentLowercase === 'e') {
+      currentLetter = 3
+   } else  if (currentLowercase === 'i') {
+       currentLetter = 1
+   } else  if (currentLowercase === 'o') {
+        currentLetter = 0
+      }
+      result.push(currentLetter)
+ }
+ return result.join('')
+}
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
-// Expected output: ["Apple", "Banana", "Orange"]
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
-
-
+describe("wordsWithLetter", () => {
+  it("takes in an array of words and a single letter and returns all the words that contain that particular letter", () => {
+    const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+    const letterA = "a"
+    const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+    const letterE = "e"
+    expect(wordsWithLetter(arrayOfWords1, letterA)).toEqual(["Apple", "Banana", "Orange"])
+    expect(wordsWithLetter(arrayOfWords2, letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+  })
+})
 
 // b) Create the function that makes the test pass.
+//iterate over array using filter
+// return value if the value includes the letter
+//use .includes method to check
+//must lowercase the value before checking for letter
+
+const wordsWithLetter = (arr, letter) => {
+  let result = arr.filter(value => {
+    if (value.toLowerCase().includes(letter) === true) {
+      return value
+    }
+  })
+  return result
+}
```

Comment:
Nice job. Filter has a built in `if` statement so you can just return the evaluation.
```
return arr.filter(value => {
    return value.toLowerCase().includes(letter) === true
}
```